infra/events/src/main/java/com/simprints/infra/events/event/local/EventDatabaseFactory.kt

```diff
@@ -1,8 +1,55 @@
 package com.simprints.infra.events.event.local
 
-internal interface EventDatabaseFactory {
-    fun build(): EventRoomDatabase
-    fun deleteDatabase()
-    fun recreateDatabaseKey()
-}
+import android.content.Context
+import com.simprints.infra.logging.Simber
+import com.simprints.infra.security.SecurityManager
+import dagger.hilt.android.qualifiers.ApplicationContext
+import net.sqlcipher.database.SQLiteDatabase.getBytes
+import net.sqlcipher.database.SupportFactory
+import javax.inject.Inject
+
+
+internal class EventDatabaseFactory @Inject constructor(
+    @ApplicationContext val ctx: Context,
+    private val securityManager: SecurityManager
+) {
+
+    fun build(): EventRoomDatabase {
+        try {
+            val key = getOrCreateKey(DB_NAME)
+            val passphrase: ByteArray = getBytes(key)
+            val factory = SupportFactory(passphrase)
+            return EventRoomDatabase.getDatabase(
+                ctx,
+                factory,
+                DB_NAME
+            )
+        } catch (t: Throwable) {
+            Simber.e(t)
+            throw t
+        }
+    }
+
+    private fun getOrCreateKey(@Suppress("SameParameterValue") dbName: String): CharArray {
+        return try {
+            securityManager.getLocalDbKeyOrThrow(dbName)
+        } catch (t: Throwable) {
+            t.message?.let { Simber.d(it) }
```

Comment:
Are we sure that we want to use debug-level logging in this scenario?

Reply:
I didn't really change that part. This file shows up in diff because I removed the redundant internal interface and this code was previously in "Impl" file.